### ISSUE TITLE
download list plugin

### ISSUE THREAD

**Issue by trob** [Fabrik/fabrik#1558]
is not working if list is ajaxfied. Sticky spinner because zip/pdf is in the network response.

BTW: I think names like
zipD12B.tmp.zip (for the zip) and
dowBF3F.tmp.pdf (for the zipped PDFs)
are not very descriptive

**Comment by cheesegrits** [comment]
Yeah, the download plugin can't work in AJAX, period.  You can't download a file through AJAX, period.  There are some nasty hacks you can do to basically reflect the AJAX call as a normal POST, but I don't want to go that route.

What we really need to do is extend the list plugin model to specify whether a plugin can use AJAX, and if not, the list.js should just post it normally.

**Comment by cheesegrits** [comment]
OK, I'm working on a way of disabling AJAX for list plugins that don't support it.

As or the file names, what would you suggest?  Can't use the filename of the file(s) being zip'ed, as there could be dozens of different files.  Are you ust suggesting that a slightly more structured name would be better, like list_x_downloads_xxxxxx.zip?

We still have to have a random component in the name, as the file has to be written out to the tmp dir for ZIP creation, can't do it in memory, so need a random part to avoid name clashes.

   -- hugh

**Comment by trob** [comment]
I've never used this plugin (didn't even know it was there ;) ).

Filenames:
yes, maybe zip containing the list name, and PDFs list name + record id?
and no ...tmp...

**Comment by pollen8** [comment]
@cheesegrits  did you fix this one?

**Comment by pollen8** [comment]
@cheesegrits  bump? otherwise I am going to close it

**Comment by cheesegrits** [comment]
Yes, I added a useAJAX() method (or some such) so list plugins that don't want it can opt out, and the download plugin does that. 

Didn't do anything about the file names as I never got a response.

**Comment by cheesegrits** [comment]
Well, I mean I'm blind and never saw the response. :)
